feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go

```diff
@@ -0,0 +1,750 @@
+	}
+	t.Logf("GetProfileResponse: %v", getProfileResponse)
+}
+
+// PostValidationCheck function to do a validation of all services after certz rotation.
+func PostValidationCheck(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
+
+	if !VerifyGnsi(t, caCert, ctx, san, serverAddr, username, password, cert) {
+		t.Fatalf("Failed with new gNSI Connection: got false, want true.")
+	}
+	t.Logf("New gNSI connection successfully completed.")
+
+	if !VerifyGnoi(t, caCert, ctx, san, serverAddr, username, password, cert) {
+		t.Logf("Failed with new gNOI Connection: got false, want true")
+		return false
+	}
+	t.Logf("New gNOI connection successfully completed.")
+
+	if !VerifyGribi(t, caCert, ctx, san, serverAddr, username, password, cert) {
+		t.Logf("Failed with new gRIBI Connection: got false, want true.")
+		return false
+	}
+	t.Logf("New gRIBI connection successfully completed.")
+	if !VerifyP4rt(t, caCert, ctx, san, serverAddr, username, password, cert) {
+		t.Logf("Failed with new P4rt Connection: got false, want true.")
+		return false
+	}
+	if !VerifyGnmi(t, caCert, ctx, san, serverAddr, username, password, cert) {
+		t.Logf("Failed with new gNMI Connection: got false, want true.")
+		return false
+	}
+	t.Logf("New gNMI connection successfully completed.")
+	return true
+}
+
+// TestNewConnection function to validate the connection for any  gRPC serviceafter successful rotation.
+func TestNewConnection(t *testing.T, caCert *x509.CertPool, ctx context.Context, san, serverAddr, username, password string, cert tls.Certificate) bool {
+
+	credOpts := []grpc.DialOption{grpc.WithBlock(), grpc.WithTransportCredentials(credentials.NewTLS(
+		&tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      caCert,
+			ServerName:   san,
+		}))}
+	creds := &rpcCredentials{&creds.UserPass{Username: username, Password: password}}
+	credOpts = append(credOpts, grpc.WithPerRPCCredentials(creds))
+	target := fmt.Sprintf("%s:%d", serverAddr, 9339)
+	conn, err := grpc.NewClient(target, credOpts...)
+	t.Logf("Err response: %v.", err)
+	if err != nil {
+		t.Logf("gRPC NewClient failed with mismatch certificate as expected to %q with error %s.", target, err)
+		return false
+	}
+	t.Logf("Connection state: %v.", conn.GetState().String())
+	defer conn.Close()
+
+	authzClient := authzpb.NewAuthzClient(conn)
+	t.Logf("Authzclient Get Request.")
+	rsp, err := authzClient.Get(ctx, &authzpb.GetRequest{})
+	t.Logf("gNSI authz get response is %s", rsp)
+	if err != nil {
+		statusError, _ := status.FromError(err)
+		if statusError.Code() == codes.FailedPrecondition {
+			t.Logf("Expected error FAILED_PRECONDITION seen for authz Get Request.")
+		} else {
+			t.Logf("Unexpected error during authz Get Request :%v.", err)
+			return false
+		}
+	}
+	conn.Close()
+	return true
+}
+
+// MismatchPostValidationCheck function to do a validation of all services after certz rotation.
```

Comment:
Can you perhaps document more clearly what the differences between the different post validation checks here are? It looks like you might be able to have something where we could just hand an argument that asks for the expected state, rather than code repetition here?